include/eve/traits/max_scalar_size.hpp

```diff
@@ -11,6 +11,19 @@
 
 namespace eve
 {
+  namespace detail
+  {
+    template<typename T> constexpr auto minmax_scalar(bool mode) noexcept
+    {
+      if constexpr(kumi::product_type<T>)
+      {
+        if(mode) return kumi::max_flat(kumi::as_tuple_t<T>{}, [](auto m) { return sizeof(m); });
+        else     return kumi::min_flat(kumi::as_tuple_t<T>{}, [](auto m) { return sizeof(m); });
+      }
+      else return sizeof(T);
```

Comment:
Why do you need this? We only use this in soa_vector, that should be kumi product

Reply:
In fact we also use it here:

```c++
/home/falcou/infra/tindalos-2/_work/eve/eve/include/eve/traits/max_scalar_size.hpp:31:67: error: constraints not satisfied for alias template 'as_tuple_t' [with Tuple = signed char, Meta = std::type_identity]
  constexpr std::size_t max_scalar_size_v = kumi::max_flat( kumi::as_tuple_t<T>{}
                                                                  ^~~~~~~~~~~~~
/home/falcou/infra/tindalos-2/_work/eve/eve/include/eve/detail/compress/simd/x86/compress_using_masks.hpp:167:32: note: in instantiation of variable template specialization 'eve::max_scalar_size_v<signed char>' requested here
    requires ( N() == 16 ) && (max_scalar_size_v<T> <= 2)
```

The old implementation used T{} instead of the as_tuple_t<T>{} that fix the issue of forcing UDT to be constexpr def-constructible. Now we need to explicitley say that we take care of scalars.

If I remove `min_scalar_size` I can however simplify the rest of the code

Reply:
Unless you tell me this is an issue in `compress_using_masks`

Reply:
Tests already do what is requried IMHO:

```c++
#include <eve/traits/max_scalar_size.hpp>

TTS_CASE( "Check max_scalar_size" )
{
  TTS_CONSTEXPR_EQUAL(sizeof(int),  eve::max_scalar_size_v<int>);
  TTS_CONSTEXPR_EQUAL(sizeof(std::int8_t), eve::max_scalar_size_v<std::int8_t>);
  TTS_CONSTEXPR_EQUAL(sizeof(std::int8_t), (eve::max_scalar_size_v<kumi::tuple<std::int8_t, std::int8_t>>));
  TTS_CONSTEXPR_EQUAL(sizeof(int),  (eve::max_scalar_size_v<kumi::tuple<
                                       int, kumi::tuple<std::int8_t, std::int8_t>>>));
};
```

Reply:
An alternative could be that `as_tuple_t<non product-type T>` in kumi returns tuple<T>.

Reply:
The more it goes, the more the kumi fix makes more sense.

Reply:
Amended kumi and simplified the fix here